lib/cext/include/ruby/encoding.h

```diff
@@ -98,7 +98,7 @@ enum ruby_coderange_type RB_ENC_CODERANGE(VALUE obj);
 #define ENC_CODERANGE_AND(a, b)      RB_ENC_CODERANGE_AND(a, b)
 #define ENCODING_CODERANGE_SET(obj, encindex, cr) RB_ENCODING_CODERANGE_SET(obj, encindex, cr)
 
-typedef struct {
+typedef struct rb_encoding {
```

Comment:
That is currently necessary because of the way the `POLYGLOT_DECLARE_STRUCT` macro is written. We'll probably add another macro that can be used with `typedef`'ed names in the near future.

Reply:
That'd be nice :smiley: 
Actually, maybe that macro could replace this one and we'd just use `POLYGLOT_DECLARE_STRUCT_NEW_NAME(struct RData)` when there is no typedef?

Reply:
The problem is, when you do `MACRO(struct RData)`, how do you strip away the `struct `, for the function name? I'm not sure if you can do that in the C preprocessor.

I was thinking more along the lines of `POLYGLOT_DECLARE_TYPE(rb_encoding)` for `typedef ... rb_encoding`, and `POLYGLOT_DECLARE_STRUCT(RData)` for `struct RData {...}`.